app/src/main/java/com/orangeelephant/sobriety/ui/screens/create/Create.kt

```diff
@@ -0,0 +1,224 @@
+                    }
+                ) {
+                    Text(text = context.getString(R.string.submit_button))
+                }
+            },
+            dismissButton = {
+                OutlinedButton(
+                    onClick = { isDialogOpen = false },
+                ) {
+                    Text(text = context.getString(R.string.cancel_button))
+                }
+            }
+        ) {
+            DatePicker(state = datePickerState)
+        }
+    }
+
+    // UI
+    Scaffold(
+        topBar = {
+            GenericTopAppBar(
+                title = R.string.create_counter,
+                scrollBehavior = scrollBehavior,
+                navigationIcon = {
+                    BackIcon(onClick = {
+                        navController.popBackStack()
+                    })
+                }
+            )
+        },
+        floatingActionButton = {
+            Row(
+                horizontalArrangement = Arrangement.SpaceBetween,
+                modifier = Modifier.padding(horizontal = 16.dp)
+            ) {
+                OutlinedButton(
+                    onClick = { navController.popBackStack() },
+                    content = {
+                        Text(text = stringResource(id = R.string.cancel_button))
+                    },
+                    shape = RoundedCornerShape(50.dp),
+                    modifier = Modifier
+                        .size(width = 170.dp, height = 50.dp)
+                )
+
+                Spacer(modifier = Modifier.width(30.dp))
+
+                Button(
+                    enabled = createConditionsMet,
+                    onClick = {
+                        val newCounter = Counter(-1,
+                            createScreenViewModel.nameText,
+                            createScreenViewModel.dateVal!!,
+                            0)
+                            createScreenViewModel.onCreateCounter(
+                                newCounter,
+                                createScreenViewModel.reasonText,
+                                onCounterCreated = { counterId ->
+                                    navController.navigate(route = Screen.CounterFullView.createRoute(counterId = counterId))
+                                })
+                              },
+                    content = {
+                        Text(text = stringResource(id = R.string.create_button))
+                    },
+                    shape = RoundedCornerShape(50.dp),
+                    modifier = Modifier
+                        .size(width = 170.dp, height = 50.dp)
+                )
+            }
+        },
+
+        floatingActionButtonPosition = FabPosition.Center
+
+    ) { innerPadding ->
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(top = innerPadding.calculateTopPadding(), bottom = 0.dp)
+        ) {
+
+            // photo
```

Comment:
are you wanting to do comments like this across the codebase because I havent been doing that, i figured it can more easily be deciphered by the variable names eg. `createScreenViewModel.selectedImageUri`

Reply:
i can remove them, its just how i code. It helps me get back into a project when ive been away for a while